clippy_lints/src/floating_point_arithmetic.rs

```diff
@@ -272,6 +274,52 @@ fn check_expm1(cx: &LateContext<'_, '_>, expr: &Expr<'_>) {
     }
 }
 
+fn is_float_mul_expr<'a>(cx: &LateContext<'_, '_>, expr: &'a Expr<'a>) -> Option<(&'a Expr<'a>, &'a Expr<'a>)> {
+    if_chain! {
+        if let ExprKind::Binary(op, ref lhs, ref rhs) = &expr.kind;
+        if let BinOpKind::Mul = op.node;
+        if cx.tables.expr_ty(lhs).is_floating_point();
+        if cx.tables.expr_ty(rhs).is_floating_point();
+        then {
+            return Some((lhs, rhs));
+        }
+    }
+
+    None
+}
+
+// TODO: Fix rust-lang/rust-clippy#4735
+fn check_fma(cx: &LateContext<'_, '_>, expr: &Expr<'_>) {
```

Comment:
NIT:
```suggestion
fn check_mul_add(cx: &LateContext<'_, '_>, expr: &Expr<'_>) {
```

Also can you rename the test files`*_fma`->`*_mul_add` for clarity.